src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/TopicItem.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Collapsible, Form, Card, Button,
+} from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { useFormikContext } from 'formik';
+import { ExpandLess, ExpandMore, Delete } from '@edx/paragon/icons';
+import messages from '../messages';
+
+const TopicItem = ({
+  intl, index, name, onDelete,
+}) => {
+  const [title, setTitle] = useState(name);
+  const [isRemove, setIsRemove] = useState(false);
```

Comment:
Why do we need this?

Reply:
Title state updates the name of the topic whenever user edits the name and  isRemove state handles the Deletion confirmation modal visibility.